src/cpp/core/r_util/RSourceIndex.cpp

```diff
@@ -366,6 +366,46 @@ void libraryCallIndexer(const RTokenCursor& cursor,
    }
 }
 
+void testThatCallIndexer(const RTokenCursor& cursor,
+                         const IndexStatus& status,
+                         bool isReadOnlyFile, 
+                         RSourceIndex* pIndex)
+{
+   if (!cursor.isType(RToken::ID))
+      return;
+   
+   if (!(cursor.contentEquals(L"test_that")))
+      return;
+   
+   RTokenCursor clone = cursor.clone();
+   if (!clone.moveToNextSignificantToken())
+      return;
+   
+   if (!clone.isType(RToken::LPAREN))
+      return;
+   
+   if (!clone.moveToNextSignificantToken())
+      return;
+   
+   if (clone.isType(RToken::STRING))
+   {
+      static const boost::regex removeQuotesRegex("^['\"](.*)['\"]$");
+
+      // the test description without the quotes, but prefixed with "t "
+      std::string desc = boost::regex_replace(clone.contentAsUtf8(), removeQuotesRegex, "t \\1");
```

Comment:
Will this handle embedded quotes correctly? E.g.

```
test_that("\"embedded quotes\" are handled", { ... })
```

Reply:
Do we need to worry about 😱 raw strings 😱 ?

Reply:
✅ the regex only deals with the first and last `'` or `"`

<img width="1555" alt="image" src="https://user-images.githubusercontent.com/2625526/167099431-07b200ec-91f0-4fa3-ac30-3054f8229572.png">